drivers/watchdog/wdt_npm6001.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_npm6001_wdt
+
+#include <errno.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/watchdog.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/toolchain.h>
+
+/* nPM6001 Watchdog related registers */
+#define NPM6001_WDARMEDVALUE	0x54U
+#define NPM6001_WDARMEDSTROBE	0x55U
+#define NPM6001_WDTRIGGERVALUE0 0x56U
+#define NPM6001_WDTRIGGERVALUE1 0x57U
+#define NPM6001_WDTRIGGERVALUE2 0x58U
+#define NPM6001_WDDATASTROBE	0x5DU
+#define NPM6001_WDPWRUPVALUE	0x5EU
+#define NPM6001_WDPWRUPSTROBE	0x5FU
+#define NPM6001_WDKICK		0x60U
+#define NPM6001_WDREQPOWERDOWN	0x62U
+
+/* nPM6001 WDTRIGGERVALUEx ms/LSB, min/max values */
+#define NPM6001_WDTRIGGERVALUE_MS_LSB 4000U
+#define NPM6001_WDTRIGGERVALUE_MIN    0x2U
+#define NPM6001_WDTRIGGERVALUE_MAX    0xFFFFFFU
+
+/* nPM6001 WDPWRUPVALUE fields */
+#define NPM6001_WDPWRUPVALUE_OSC_ENABLE	    BIT(0)
+#define NPM6001_WDPWRUPVALUE_COUNTER_ENABLE BIT(1)
+#define NPM6001_WDPWRUPVALUE_LS_ENABLE	    BIT(2)
+
+struct wdt_npm6001_config {
+	struct i2c_dt_spec bus;
+};
+
+static int wdt_npm6001_setup(const struct device *dev, uint8_t options)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(options);
+
+	return 0;
+}
+
+static int wdt_npm6001_disable(const struct device *dev)
+{
+	const struct wdt_npm6001_config *config = dev->config;
+	uint8_t buf[4] = {NPM6001_WDARMEDVALUE, 1U, NPM6001_WDARMEDSTROBE, 1U};
+
```

Comment:
Same here, perhaps?

Reply:
I've just checked and wdt API doesn't document `-EWOULDBLOCK` or `isr-ok`, we should likely update the interface first as it doesn't seem to be a requirement now.